grafana-plugin/src/containers/UserSettings/parts/tabs/GoogleCalendar/GoogleCalendar.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from 'react';
+
+import { Button, Divider, HorizontalGroup, InlineSwitch, VerticalGroup } from '@grafana/ui';
+import { observer } from 'mobx-react';
+
+import { Block } from 'components/GBlock/Block';
+import { Text } from 'components/Text/Text';
+import { WithConfirm } from 'components/WithConfirm/WithConfirm';
+import { GSelect } from 'containers/GSelect/GSelect';
+import { WithPermissionControlTooltip } from 'containers/WithPermissionControl/WithPermissionControlTooltip';
+import GoogleLogo from 'icons/GoogleLogo';
+import { Schedule } from 'models/schedule/schedule.types';
+import { UserHelper } from 'models/user/user.helpers';
+import { ApiSchemas } from 'network/oncall-api/api.types';
+import { useStore } from 'state/useStore';
+import { UserActions } from 'utils/authorization/authorization';
+
+const GoogleCalendar: React.FC<{ id: ApiSchemas['User']['pk'] }> = observer(({ id }) => {
+  const { userStore, scheduleStore } = useStore();
+
+  const user = userStore.items[id];
+  const [googleCalendarSettings, setGoogleCalendarSettings] = useState(user?.google_calendar_settings);
+  const [showSchedulesDropdown, setShowSchedulesDropdown] = useState(
+    user.google_calendar_settings?.specific_oncall_schedules_to_sync.length > 0
+  );
+
+  const handleShowSchedulesDropdownChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.checked;
+    setShowSchedulesDropdown(value);
+
+    if (!value) {
+      handleSchedulesChange([]);
+    }
+  };
+
+  useEffect(() => {
+    if (user) {
+      setGoogleCalendarSettings(user.google_calendar_settings);
+    }
+  }, [user]);
+
+  const handleSchedulesChange = (value) => {
+    setGoogleCalendarSettings((v) => ({ ...v, specific_oncall_schedules_to_sync: value }));
+
+    userStore.updateCurrentUser({
+      google_calendar_settings: { ...googleCalendarSettings, specific_oncall_schedules_to_sync: value },
+    });
+  };
```

Comment:
Do we need to have local state for google calendar settings + the useEffect that updates this local state on user updates? Google calendar settings is a part of current user, current user comes from MobX and is an observable. So we should be able to omit local state and useEffect by just updating user in store + consuming google calendar settings from store user